api/src/main/java/net/kyori/adventure/text/Component.java

```diff
@@ -1872,6 +1874,47 @@ default boolean hasStyling() {
   @Contract(pure = true)
   @NotNull Component replaceText(final @NotNull TextReplacementConfig config);
 
+  /**
+   * Returns an iterable view of this component.
+   *
+   * @param type the type
+   * @return the iterable
+   * @see ComponentIteratorType
+   * @since 4.8.0
+   */
+  default @NotNull Iterable<Component> iterable(final @NotNull ComponentIteratorType type) {
+    Objects.requireNonNull(type, "type");
+    return new ForwardingIterator<>(() -> this.iterator(type), () -> this.spliterator(type));
+  }
+
+  /**
+   * Returns an iterator for this component.
+   *
+   * <p>As components are immutable, this iterator does not support removal.</p>
+   *
+   * @param type the type of the iterator
+   * @return the iterator
+   * @see ComponentIteratorType
+   * @since 4.8.0
+   */
+  default @NotNull Iterator<Component> iterator(final @NotNull ComponentIteratorType type) {
+    return new ComponentIterator(this, type);
+  }
+
+  /**
+   * Returns a spliterator for this component.
+   *
+   * <p>The resulting spliterator has the {@link Spliterator#IMMUTABLE}, {@link Spliterator#NONNULL} and {@link Spliterator#ORDERED} characteristics.</p>
+   *
+   * @param type the type of the underlying iterator
+   * @return the iterator
+   * @see ComponentIteratorType
+   * @since 4.8.0
+   */
+  default @NotNull Spliterator<Component> spliterator(final @NotNull ComponentIteratorType type) {
+    return Spliterators.spliteratorUnknownSize(this.iterator(type), Spliterator.IMMUTABLE & Spliterator.NONNULL & Spliterator.ORDERED);
+  }
```

Comment:
Do we want to move these methods to their own class instead, and access them statically?

Reply:
I think that is unnecessary -- they're one-liners in here already.